### PR TITLE
Wizard recipe: NetCDFF-v4.5.3

### PR DESCRIPTION
**Pull request by ctessum**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: NetCDFF
* Version: v4.5.3

@staticfloat please review and merge.

### CODE REVIEW

File: N/NetCDFF/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libnetcdff", :libnetcdff)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="NetCDF_jll", uuid="7243133f-43d8-5620-bbf4-c2c921802cf3"))
```
Comment (giordano) [inline N/NetCDFF/build_tarballs.jl:36]: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=9914&view=logs&j=9bfd259a-c2a1-53f6-dea5-4be155f9c940&t=8329a79a-e40d-5680-2144-5971d3335d88&l=1207
```
┌ Warning: To ensure that the correct version of libgfortran is found at runtime, add the following entry to the list of dependencies of this builder
│ 
│     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/EyOxg/src/auditor/compiler_abi.jl:4
```

```suggestion
    Dependency(PackageSpec(name="NetCDF_jll", uuid="7243133f-43d8-5620-bbf4-c2c921802cf3")),
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
```

File: N/NetCDFF/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("x86_64", "macos"; )
+]
+
```
Comment (giordano) [inline N/NetCDFF/build_tarballs.jl:27]: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=9914&view=logs&j=9bfd259a-c2a1-53f6-dea5-4be155f9c940&t=8329a79a-e40d-5680-2144-5971d3335d88&l=1215
```
┌ Warning: /agent/_work/1/s/N/NetCDFF/build/x86_64-linux-gnu/M0dgUNVX/x86_64-linux-gnu-libgfortran3-cxx03/destdir/lib/libnetcdff.so.7.0.0 links to libgfortran!  This causes incompatibilities across major versions of GCC.  To remedy this, you must build a tarball for each major version of GCC.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_gfortran_versions(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/EyOxg/src/auditor/compiler_abi.jl:60
```
```suggestion
platforms = expand_gfortran_versions(platforms)
```

**Comment by ctessum** [comment]
Thanks!

**Comment by giordano** [comment]
@ctessum any particular reason why only these two platforms?

**Comment by ctessum** [comment]
No particular reason, other than lack of experience with the ecosystem. I think it is at least limited to the [platforms supported by NetCDF](https://github.com/JuliaPackaging/Yggdrasil/blob/master/N/NetCDF/common.jl#L57), but I was having difficulty getting the windows build to work, so I gave up after these 2.

**Comment by giordano** [comment]
Ok, let me see what's the issue with the other platforms, if I can't quickly figure out the issue we'll merge the current version

**Comment by ctessum** [comment]
Thanks!

**Comment by giordano** [comment]
Thanks for your contribution!

**Comment by ctessum** [comment]
Thank you for your help!

**Pull request merged**